packages/manager/apps/pci-load-balancer/src/components/detail/listeners/l7/PolicyForm.component.tsx

```diff
@@ -0,0 +1,434 @@
+    projectId,
+    region,
+    loadBalancerId,
+  });
+
+  const filteredPools = useMemo(() => {
+    if (pools && listener) {
+      return pools?.filter(
+        (pool) =>
+          !UNAVAILABLE_POOL_PROTOCOLS.includes(pool.protocol) &&
+          LISTENER_POOL_PROTOCOL_COMBINATION[listener?.protocol].includes(
+            pool.protocol,
+          ),
+      );
+    }
+    return [];
+  }, [pools, listener]);
+
+  const [isTouchedName, setIsTouchedName] = useState(false);
+  const [isTouchedPrefix, setIsTouchedPrefix] = useState(false);
+  const [isTouchedURL, setIsTouchedURL] = useState(false);
+  const [hasErrorName, setHasErrorName] = useState(false);
+
+  useEffect(() => {
+    setHasErrorName(isTouchedName && policyState.name === '');
+  }, [policyState.name, isTouchedName]);
+
+  const isPending = isPendingListener || isPendingPools || isPendingAction;
+
+  const onActionChange = (
+    event: OsdsSelectCustomEvent<OdsSelectValueChangeEventDetail>,
+  ) => {
+    const action = event.detail.value as string;
+    const policyUpdate = {
+      ...policyState,
+      action,
+    };
+    if ([ACTIONS.REDIRECT_TO_URL, ACTIONS.REDIRECT_PREFIX].includes(action)) {
+      policyUpdate.redirectHttpCode = 302;
+    } else {
+      policyUpdate.redirectHttpCode = undefined;
+    }
+    if (action !== ACTIONS.REDIRECT_TO_URL) {
+      policyUpdate.redirectUrl = undefined;
+      setIsTouchedURL(false);
+    }
+    if (action !== ACTIONS.REDIRECT_PREFIX) {
+      policyUpdate.redirectPrefix = undefined;
+      setIsTouchedPrefix(false);
+    }
+    setPolicyState(policyUpdate);
+    onChange(policyUpdate);
+  };
+
+  const prefixError = useMemo(() => {
+    if (isTouchedPrefix) {
+      if (!policyState.redirectPrefix) {
+        return tPciCommon('common_field_error_required');
+      }
+      if (!RegExp(URL_PATTERN).test(policyState.redirectPrefix)) {
+        return tPciCommon('common_field_error_pattern');
+      }
+    }
+    return '';
+  }, [isTouchedPrefix, policyState.redirectPrefix]);
+
+  const isDisabled = useMemo(() => {
+    switch (policyState.action) {
+      case ACTIONS.REDIRECT_TO_URL:
+        return (
+          !policyState.name ||
+          !RegExp(URL_PATTERN).test(policyState.redirectUrl)
+        );
+      case ACTIONS.REDIRECT_PREFIX:
+        return (
+          !policyState.name ||
+          !RegExp(URL_PATTERN).test(policyState.redirectPrefix)
+        );
+      case ACTIONS.REJECT:
+        return !policyState.name;
+      case ACTIONS.REDIRECT_TO_POOL:
+        return !policyState.name || !policyState.redirectPoolId;
+      default:
+        return true;
+    }
+  }, [policyState]);
+
+  return (
+    <>
+      {isPending && <OsdsSpinner size={ODS_SPINNER_SIZE.md} inline />}
+      {!isPending && (
+        <div>
+          <OsdsFormField
+            error={
+              hasErrorName ? tPciCommon('common_field_error_required') : ''
+            }
+          >
+            <OsdsText
```

Comment:
A LabelComponent created by @sidlynx can be used here, which would be helpful in standardizing the formFieldLabel across the entire application.